aptos-move/aptos-vm-types/src/change_set.rs

```diff
@@ -185,66 +195,115 @@ impl VMChangeSet {
     ) {
-        self.aggregator_write_set
+        self.aggregator_v1_write_set
             .extend(additional_aggregator_writes)
     }
 
     pub fn aggregator_v1_write_set(&self) -> &HashMap<StateKey, WriteOp> {
-        &self.aggregator_write_set
+        &self.aggregator_v1_write_set
     }
 
     pub fn aggregator_v1_delta_set(&self) -> &HashMap<StateKey, DeltaOp> {
-        &self.aggregator_delta_set
+        &self.aggregator_v1_delta_set
+    }
+
+    pub fn aggregator_v2_change_set(&self) -> &HashMap<AggregatorID, AggregatorChange> {
+        &self.aggregator_v2_change_set
     }
 
     pub fn events(&self) -> &[ContractEvent] {
         &self.events
     }
 
-    /// Materializes this change set: all deltas are converted into writes and
-    /// are combined with existing aggregator writes.
-    pub fn try_materialize(self, state_view: &impl StateView) -> anyhow::Result<Self, VMStatus> {
+    /// Materializes this change set: all aggregator v1 deltas are converted into writes and
+    /// are combined with existing aggregator writes. The aggregator v2 changeset is not touched.
+    pub fn try_materialize_aggregator_v1_delta_set(
+        self,
+        state_view: &impl StateView,
+    ) -> anyhow::Result<Self, VMStatus> {
         let Self {
             resource_write_set,
             module_write_set,
-            mut aggregator_write_set,
-            aggregator_delta_set,
+            mut aggregator_v1_write_set,
+            aggregator_v1_delta_set,
+            aggregator_v2_change_set,
             events,
         } = self;
 
         let into_write =
-            |(state_key, delta): (StateKey, DeltaOp)| -> anyhow::Result<(StateKey, WriteOp), VMStatus> {
-                let write = delta.try_into_write_op(state_view, &state_key)?;
+            |(state_key, aggregator_change): (StateKey, DeltaOp)| -> anyhow::Result<(StateKey, WriteOp), VMStatus> {
+                let write = aggregator_change.try_into_write_op(state_view, &state_key)?;
                 Ok((state_key, write))
             };
 
-        let materialized_aggregator_delta_set =
-            aggregator_delta_set
-                .into_iter()
-                .map(into_write)
-                .collect::<anyhow::Result<HashMap<StateKey, WriteOp>, VMStatus>>()?;
-        aggregator_write_set.extend(materialized_aggregator_delta_set.into_iter());
+        let materialized_aggregator_delta_set = aggregator_v1_delta_set
+            .into_iter()
+            .map(into_write)
+            .collect::<anyhow::Result<HashMap<StateKey, WriteOp>, VMStatus>>()?;
+        aggregator_v1_write_set.extend(materialized_aggregator_delta_set.into_iter());
+
+        Ok(Self {
+            resource_write_set,
+            module_write_set,
+            aggregator_v1_write_set,
+            aggregator_v1_delta_set: HashMap::new(),
+            aggregator_v2_change_set,
+            events,
+        })
+    }
+
+    pub fn try_materialize_aggregator_v2_changes(
```

Comment:
I started a discussion about this (not sure how to handle txn simulation the best way), but let's delete this method for now and do it in separate PRs.